<div class="alert alert-block alert-info">
Singapore Management University<br>
CS105 Statistical Thinking for Data Science, 2023/24 Term 2
</div>

# CS105 Group Project Submission (Part I)

-----
Provide your team details, including section, team number, team members, and the name of the dataset. 
Complete all of the following sections. For any part requiring code to derive your answers, please create a code cell immediately below your response and run the code.
To edit any markdown cell, double click the cell; after editing, execute the markdown cell to collapse it.
<br>
-----

## Declaration

<span style="color:red">By submitting this notebook, we declare that **no part of this submission is generated by any AI tool**. We understand that AI-generated submissions will be considered as plagiarism, and just like other plagirisum cases, disciplinary actions will be imposed.</span>

#### Section: G1
#### Team: Group 1
#### Members:
1. CHONG SHIH RUI ZACHARY
2. YANG ZHONG HNG REUBEN
3. KEAGAN PAUL POULOSE
4. LUCAS SONG KANGZHENG
5. 

#### Dataset: Wine dataset

## Part I: Exploratory Data Analysis (EDA) [8% of final grade]

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
df = pd.read_csv('wine.csv')

In [4]:
df.head()

ID  Fixed_Acidity  Volatile_Acidity  Citric_Acid Residual_Sugar  Chlorides  \
0   1            7.0              0.27         0.36           High      0.045   
1   2            6.3              0.30         0.34            Low      0.049   
2   3            8.1              0.28         0.40            Low      0.050   
3   4            7.2              0.23         0.32       Moderate      0.058   
4   5            7.2              0.23         0.32       Moderate      0.058   

   Free_SO2  Total_SO2    Density      pH  Sulphates Alcohol  Quality  
0      45.0      170.0      >1.00  Acidic       0.45      8%        6  
1      14.0      132.0  0.99-1.00  Acidic       0.49      9%        6  
2      30.0       97.0  0.99-1.00  Acidic       0.44     10%        6  
3      47.0      186.0  0.99-1.00  Acidic       0.40      9%        6  
4      47.0      186.0  0.99-1.00  Acidic       0.40      9%        6

### 1. Overview of dataset [15% of Part I]

**a.** Summarise the background of the dataset [limited to 50 words]

**This dataset gives the quality of wine.
Each row is an observation and represents a type of wine, and along each row, you find values that describe certain attributes about the wine. In particular, the variable Quality indicates whether the quality of the wine leans more towards good or bad.**

**b.** State the size of the dataset

In [10]:
df.size

63674

**63674** 

**c.** For each variable, describe what it represents and its data type (numerical or categorical)

|Column name|Description|Data type|
|---|---|---|
|ID| Unique Id of a wine | Categorical |
|Fixed_Acidity| Amount of Tartaric acid, measured in g/dm3 | Numerical |
|Volatile_Acidity| Amount of Acetic acid, measured in g/dm3  | Numerical |
|Citric_Acid| Amount measured in g/dm3 | Numerical |
|Residual_Sugar| Level of residual sugar content | Categorical (“High”, “Moderate”, “Low”) |
|Chlorides| Amount of Sodium chloride, measured in g/dm3 | Numerical |
|Free_SO2| Amount measured in mg/dm3 | Numerical |
|Total_SO2| Amount measured in mg/dm3  | Numerical |
|Density| Amount measured in g/dm3 | Categorical (“<0.99”, “0.99-1.00”, “>1.00”) |
|pH| The measure of acidity of the wine | Categorical (“Neutral”, “Acidic”, “Highly Acidic”)| 
|Sulphates|Potassium sulphate, measured in g/dm3 | Numerical |
|Alcohol| Level of alcohol content | Categorical (8% through 14% in discrete levels)|
|Quality| Score based on sensory data | Categorical (0 through 10 (0 is bad, 10 is good))|

### 2. Data pre-processing [35% of Part I]

**a.** For each variable, determine the percentage of missing data. For any column with missing data, describe how you resolve the issue. Clearly state any assumption you made.

**Response.** 

**b.** For each variable, identify outliers (if any) and describe how you resolve the issue. Clearly state any assumption you made.

**Response.** 

**c.** For categorical variables, perform the necessary encoding.

**Response.** 

### 3.	Exploratory analysis and visualization [50% of Part I]

**a.** For each variable, provide relevant summary statistics

**Response.** 

**b.** For each variable, provide an appropriate visualisation depicting the distribution of its values, and summarize any key observation(s) you made.

**Response.** 

**c.** Perform bi-variate analysis on the variables. You do not need to present the analysis of every pair of variables; only focus on the pairs you believe are worth investigating and explain. For each pair, describe the relationship between the two variables. Use appropriate statistical methods and/or visualizations.

**Response.** 